### PR TITLE
feat(RHICOMPL-689): Implement support for multi-ssg-version policies

### PR DESCRIPTION
**Pull request by bastilian**
This implements support for policies with multiple SSG versions:

Screenshot 2020-10-29 at 12 59 08

Enable via: /insights/compliance/scappolicies/POLICY_ID?multiversionTabs=enable

Note: There are issues with using other components in a Patternfly `Tab` title where even `!important` properties are ignored

### CODE REVIEW

**Review by akofink — APPROVED** [review]
Works as expected. Thanks, @bastilian!

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { SystemRulesTable, selectRulesTableColumns } from '@redhat-cloud-services/frontend-components-inventory-compliance';
+import { RoutedTabs } from 'PresentationalComponents';
+
+const PolicyMultiversionRules = ({ policy: { policy: { profiles } } }) => (
+    <React.Fragment>
+        <Alert variant="info" isInline title="Rule editing coming soon" />
+        <PageSection variant={ PageSectionVariants.light }>
+            <RoutedTabs level={ 1 } defaultTab={ profiles[0].ssgVersion }>
+                {
+                    profiles.map((profile) => (
```
Comment (vkrizan) [inline src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js:13]: What's the order of the profiles / tabs?

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                                    }}>{ profile.rules.length }</Label>
+                                </>
+                            }
+                            eventKey={ profile.ssgVersion }>
+                            <SystemRulesTable
+                                remediationsEnabled={false}
+                                columns={ selectRulesTableColumns(['Rule', 'Severity', 'Ansible']) }
+                                profileRules={[{
+                                    profile: { ...profile },
+                                    rules: profile.rules
```
Comment (vkrizan) [inline src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js:32]: A bit unrelated: The Ansible is shown as "No" for rules on other profiles/SSGs

Screenshot_2020-10-30 Compliance

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                    profiles.map((profile) => (
+                        <Tab
+                            key={ profile.ssg_version }
+                            title={
+                                <>
+                                    SSG { profile.ssgVersion }
+                                    <Label color="blue" style={{
+                                        marginRight: '.5rem !important',
+                                        backgroundColor: '#1C66CB !important',
+                                        color: '#FFFFFF !important'
```
Comment (vkrizan) [inline src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js:22]: Observation: The color and gutter values could come from variables.

File: src/PresentationalComponents/TabSwitcher/TabSwitcher.js [hunk 4]
```diff
@@ -17,8 +28,8 @@ TabSwitcher.propTypes = {
     children: propTypes.node
 };
 
-export const RoutedTabSwitcher = ({ children, defaultTab }) => {
-    const currentAnchor = useAnchor(defaultTab);
+export const RoutedTabSwitcher = ({ children, defaultTab, level }) => {
+    const { currentAnchor } = useAnchorLevels(defaultTab, level);
```
Comment (vkrizan) [inline src/PresentationalComponents/TabSwitcher/TabSwitcher.js:32]: I wonder if there is an official way on how to handle deeper tabs selection via URL.

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js (hunk 1 above)
Comment (bastilian) [reply src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js:13]: As they are returned from the backend.

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js (hunk 2 above)
Comment (bastilian) [reply src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js:32]: Yep. This has been like this from the start. The mockups do not show any no state yet, but we could look into replacing it. There is a `<Ansible />` component in the frontend-components repo that may help.

File: src/PresentationalComponents/TabSwitcher/TabSwitcher.js (hunk 4 above)
Comment (bastilian) [reply src/PresentationalComponents/TabSwitcher/TabSwitcher.js:32]: Well, there is no "official" way to use the anchors for Tab delegation at all.

**Comment by vkrizan** [comment]
There is a weird behavior when coming from the policies table into the details, where title, breadcrumbs and "Policy Type" changes when after the GQL request is complete.

I have a policy named "My CIS". On the table I see "CIS Red Hat Enterprise Linux 7 Benchmark" and when I load it (click) then it keeps showing the "CIS Red Hat Enterprise Linux 7 Benchmark" for a while and then it swaps it to "My CIS".

**Comment by bastilian** [comment]
@vkrizan Thank you for testing this and catching that. However, this seems to already also happen on CI without this feature even being available. I suggest we open a Jira and look into this separately

Kapture 2020-10-30 at 10 20 20

**Review by vkrizan — APPROVED** [review]

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js (hunk 1 above)
Comment (vkrizan) [reply src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js:13]: Which means that they are unordered. How they should be ordered? The mockups show descending order.

File: src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js (hunk 3 above)
Comment (bastilian) [reply src/SmartComponents/PolicyDetails/PolicyMultiversionRules.js]: These were not working at all actually. Patternfly strips some styles under certain conditions, even with !important. Just using "blue" as the color looks fine for now, it's not the colours we want, but we'll get those when we can resolve the issues with custom styling.

**Pull request merged**